nixos/modules/services/networking/nat.nix

```diff
@@ -9,7 +9,16 @@ with lib;
 let
   cfg = config.networking.nat;
 
-  dest = if cfg.externalIP == null then "-j MASQUERADE" else "-j SNAT --to-source ${cfg.externalIP}";
+  mkDest = externalIP: if externalIP == null
+                       then "-j MASQUERADE"
+                       else "-j SNAT --to-source ${externalIP}";
+  dest = mkDest cfg.externalIP;
+  destIPv6 = mkDest cfg.externalIPv6;
+
+  enableIPv6 = config.networking.enableIPv6 && cfg.enableIPv6;
```

Comment:
I think this might be better as an assertion, one that checks that `networking.enableIPv6` is enabled when `enableIPv6` is.